### PR TITLE
chore: update typescript

### PR DESCRIPTION
**Pull request by ZachJW34**
### User facing changelog
n/a

### Additional details
Update to latest version of typescript. Had to bump `ts-loader` as `@packages/extension` was failing with the new version of TS.

### Steps to test
Counting on CI here, everything worked well locally.

### How has the user experience changed?
na

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/scaffold-config/src/frameworks.ts [hunk 1]
```diff
@@ -61,11 +61,9 @@ function getBundlerDependency (bundler: WizardBundler['type'], projectPath: stri
   }
 }
 
-export const WIZARD_MOUNT_MODULES = ['cypress/react', 'cypress/react18', 'cypress/vue', 'cypress/vue2', 'cypress/angular'] as const
+export type WizardMountModule = Awaited<ReturnType<typeof WIZARD_FRAMEWORKS[number]['mountModule']>>
```
Comment (ZachJW34) [inline packages/scaffold-config/src/frameworks.ts:64]: Example using the new `Awaited` utility type.

File: packages/driver/src/dom/elements/nativeProps.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ const descriptor = <T extends keyof Window, K extends keyof Window[T]['prototype
   const desc = Object.getOwnPropertyDescriptor(window[klass].prototype, prop)
 
   if (desc === undefined) {
-    throw new Error(`Error, could not get property descriptor for ${klass}  ${prop}. This should never happen`)
+    throw new Error(`Error, could not get property descriptor for ${klass}  ${prop as string}. This should never happen`)
```
Comment (lmiller1990) [inline packages/driver/src/dom/elements/nativeProps.ts:15]:
```suggestion
    throw new Error(`Error, could not get property descriptor for ${klass}  ${String(prop)}. This should never happen`)
```

File: packages/driver/src/dom/elements/nativeProps.ts [hunk 3]
```diff
@@ -238,7 +238,7 @@ export const getNativeProp = function<T, K extends keyof T> (obj: T, prop: K): T
   if (!nativeProp) {
     const props = _.keys(nativeGetters).join(', ')
 
-    throw new Error(`attempted to use a native getter prop called: ${prop}. Available props are: ${props}`)
+    throw new Error(`attempted to use a native getter prop called: ${prop as string}. Available props are: ${props}`)
```
Comment (lmiller1990) [inline packages/driver/src/dom/elements/nativeProps.ts:241]:
```suggestion
    throw new Error(`attempted to use a native getter prop called: ${String(prop)}. Available props are: ${props}`)
```

File: packages/driver/src/dom/elements/nativeProps.ts [hunk 4]
```diff
@@ -258,7 +258,7 @@ export const setNativeProp = function<T, K extends keyof T> (obj: T, prop: K, va
   if (!nativeProp) {
     const fns = _.keys(nativeSetters).join(', ')
 
-    throw new Error(`attempted to use a native setter prop called: ${prop}. Available props are: ${fns}`)
+    throw new Error(`attempted to use a native setter prop called: ${prop as string}. Available props are: ${fns}`)
```
Comment (lmiller1990) [inline packages/driver/src/dom/elements/nativeProps.ts:261]:
```suggestion
    throw new Error(`attempted to use a native setter prop called: ${String(prop)}. Available props are: ${fns}`)
```

File: packages/scaffold-config/src/frameworks.ts (hunk 1 above)
Comment (lmiller1990) [reply packages/scaffold-config/src/frameworks.ts:64]: 🎉

Comment (lmiller1990) [reply packages/scaffold-config/src/frameworks.ts:64]: For fun, if you do "go to definition" on `Awaited` you can see how they built it:

```ts
type Awaited<T> =
    T extends null | undefined ? T : // special case for `null | undefined` when not in `--strictNullChecks` mode
        T extends object & { then(onfulfilled: infer F): any } ? // `await` only unwraps object types with a callable `then`. Non-object types are not unwrapped
            F extends ((value: infer V, ...args: any) => any) ? // if the argument to `then` is callable, extracts the first argument
                Awaited<V> : // recursively unwrap the value
                never : // the argument to `then` was not callable
        T; // non-object or non-thenable
```

**Review by lmiller1990 — APPROVED** [review]
Nice, couple of minor improvements, won't block though.

File: packages/scaffold-config/src/frameworks.ts (hunk 1 above)
Comment (ZachJW34) [reply packages/scaffold-config/src/frameworks.ts:64]: 🤯 would take me a whole day to try and cook up something like that from scratch

**Comment by emilyrohrbough** [comment]
Should we go ahead and update the typescript versions in our system tests?

**Comment by ZachJW34** [comment]
@emilyrohrbough I wouldn't want to update the system-tests since they were created to represent a project in a given state. For example, `angular-13` has `"typescript": "~4.6.2"` and might not work with a bump. There are probably a few I could update but not sure if there is any improvement in doing so.

**Comment by emilyrohrbough** [comment]
Gottchya 👍🏻  is there a strategy on knowing when it's appropriate to bumping these system-test dependencies for component testings? or not really?
I had assumed they were the version used when added and we just haven't necessarily been bumping deps due to other work

**Review by emilyrohrbough — APPROVED** [review]

**Comment by ZachJW34** [comment]
We don't have a strategy, but for component testing I'd say it would make sense to simply remove system tests once we drop support for that version, so if we drop `angular@13` support or `nuxt@2` then they should be deleted.

File: packages/scaffold-config/src/frameworks.ts (hunk 1 above)
Comment (lmiller1990) [reply packages/scaffold-config/src/frameworks.ts:64]: It's quite cool, notice they use `Awaited` in the definition? Recursive types!

This is actually a great type to study... you can learn

- extends
- conditionals
- infer
- recursive types
- never (equivalent of `throw Error('this is impossible')` for types

**Pull request merged**